### DISCUSSION TITLE
parameters to tune for applications with a lot of objects per image

### DISCUSSION THREAD

**Discussion by c-axel** [facebookresearch/detectron2#2799] [Q&A]
I'm running mask_rcnn_R_50_FPN_3x on 1024x1024 drone images of solar panels. it is not uncommon for a single image to have 250-300ish panels. i have changed the cfg.TEST.DETECTIONS_PER_IMAGE to 400 which is more than i would expect to ever see in a single image.

the model does pretty good, but occasionally misses panels that should be very easy detections and look nearly identical to successfully detected neighboring panels. are there any other parameters I should be tweaking for images with hundreds of true objects in them? do I need to change any RPN TOPK or positive fraction parameters or anything like that?

thanks

**Comment by c-axel** [comment]
For anyone interested, through experimentation I was able to resolve my problem by changing to the following:
  cfg.MODEL.RPN.POST_NMS_TOPK_TEST = 2000
  cfg.MODEL.RPN.PRE_NMS_TOPK_TEST = 2000
  cfg.MODEL.ROI_HEADS.POSITIVE_FRACTION = 0.5
  
  would still be curious to hear input from someone who knows what they are doing tho

**Reply by MiXaiLL76** [reply]
https://github.com/facebookresearch/detectron2/issues/1481
